filament/src/RenderPass.cpp

```diff
@@ -669,7 +669,9 @@ RenderPass::Command* RenderPass::generateCommandsImpl(uint32_t extraFlags,
 
                 // FIXME: should writeDepthForShadowCasters take precedence over mi->getDepthWrite()?
                 cmdDepth.primitive.rasterState.depthWrite = (1 // only keep bit 0
-                        & (mi->isDepthWriteEnabled() | (mode == TransparencyMode::TWO_PASSES_ONE_SIDE))
+                        & (mi->isDepthWriteEnabled()
+                            || (mode == TransparencyMode::TWO_PASSES_ONE_SIDE)
+                            || Variant::isPickingVariant(variant))
```

Comment:
Should I add a boolean material property to support excluding a material from picking?

Reply:
@pixelflinger can explain this in more detail but this function is in the deep inner loop of Filament and written to be branchless. You should be using long circuit operators instead of short circuit. I'm also not sure if we want to test for picking here.

Reply:
> You should be using long circuit operators instead of short circuit.

Then I can cast the boolean value to an integer to avoid the compilation error.

> I'm also not sure if we want to test for picking here.

It is required to render the transparent objects in the picking pass.